loglist2/logfilter.go

```diff
@@ -85,11 +93,14 @@ func (ll *LogList) RootCompatible(cert *x509.Certificate, certRoot *x509.Certifi
 	return compatible
 }
 
-// TemporalCompatible creates a new LogList containing only the logs of
+// TemporallyCompatible creates a new LogList containing only the logs of
 // original LogList that are compatible with the provided cert, according to
 // NotBefore and TemporalInterval matching.
-func (ll *LogList) TemporalCompatible(cert *x509.Certificate) LogList {
+func (ll *LogList) TemporallyCompatible(cert *x509.Certificate) LogList {
 	var compatible LogList
+	if cert == nil {
+		return compatible
```

Comment:
I wonder whether this could lead to some tricky-to-discover bugs, where it's not obvious that a nil cert is being passed to this method? Should at least document this behaviour and add a test for it.